added xref for more CLI commands

jc-berger opened the pull request:
Please see the [GitHub Issue #2722](https://github.com/openshift/odo/issues/2722url)

@openshift/team-documentation and  @peer-review-squad , please review. 

@boczkowska , tagging for your awareness as well. 

Please note, I do not believe the xref that I included links properly. This being my first time incorporating an xref, I couldn't figure out why. So I thought I'd make the PR, get feedback on the xref, and feedback on any other suggested changes and recommendations. 

Also, it appears I do not have the capability to add labels to issues.

bmcelvee commented:
Congrats on your first PR, @jc-berger! 

Just want to confirm a few things while I'm building the preview locally: is https://github.com/openshift/odo/issues/2722 the correct issue? It was a bit wonky for me when I tried to open the link. 

And to which versions does this PR need to be cherry picked?

bmcelvee commented on cli_reference/developer_cli_odo/understanding-odo.adoc:
```diff
@@ -121,3 +121,4 @@ wildfly     openshift     10.0,10.1,11.0,12.0,13.0,14.0,15.0,16.0,17.0,18.0,19.0
 ----
 +
 The `TAGS` column represents the available image versions, for example, `10` represents the `rhoar-nodejs/nodejs-10` container image.
+If you want to learn more about the CLI commands, go to xref:../../cli_reference/developer_cli_odo/odo-cli-reference.adoc [odo CLI reference].
```
S/If you want to learn more.../To learn more...

And for the link, give this a try: 
xref:../../cli_reference/developer_cli_odo/odo-cli-reference.adoc#odo-cli-reference[odo CLI reference]

bmcelvee replied:
Thanks! I'll check it again and make sure the link works in a few minutes.

jc-berger replied:
Comments addressed, so I'm resolving this conversation.

jc-berger commented:
@bmcelvee, thanks for reviewing this for me. 

Yes,  https://github.com/openshift/odo/issues/2722 is the correct issue. I'm not sure why it's not building...maybe because the issue is closed? 

Also, what do you mean by `which version` this PR is for?

bmcelvee commented:
> @bmcelvee, thanks for reviewing this for me.
> 
> Yes, [openshift/odo#2722](https://github.com/openshift/odo/issues/2722) is the correct issue. I'm not sure why it's not building...maybe because the issue is closed?

That's okay! I just wanted to make sure I was looking at the correct issue. :)
> 
> Also, what do you mean by `which version` this PR is for?

This PR will merge to master, then we will cherry pick it to the appropriate versions (e.g., enterprise-4.6, enterprise-4.5, enterprise-4.4, enterprise-4.3). But before that, once we determine the appropriate versions that this change applies to, we will set the labels to reflect those version and the milestone (for enterprise-4.6, it would be future release; for enterprise-4.6 and earlier versions, it would be next release). Does that make sense?

jc-berger commented:
@bmcelvee, thanks Brandi for you patience and answering my questions :) 

Yes, that makes sense, and this fix will be for version OpenShift 4.6 Async

bmcelvee commented:
I'm happy to help! :) 

Don't forget the `#odo-cli-reference` in the xref (it's the anchor for the link); and once it's all set with the link working and the build is correct, it'll be time to squash commits.

vikram-redhat commented:
> Yes, [openshift/odo#2722](https://github.com/openshift/odo/issues/2722) is the correct issue. I'm not sure why it's not building...maybe because the issue is closed?

There is a `url` added to the link in the first comment. That is why the link to the issue is not working. 

The build is not working because the actual anchor is missing: 

`Unknown ID or title "odo-cli-reference-1", used as an internal cross reference`. As @bmcelvee mentioned, the anchor needs to be added.

jc-berger commented:
@bmcelvee and @vikram-redhat, thank you both for the help and review. 

I have squashed the commits, and you can see the revisions in the new, squashed commit, `added xref for more CLI commands` 

Thanks!

jc-berger commented:
@bmcelvee, I'm following up regarding the review of the squashed commit. This PR doesn't need to be merged until October 29th, so no rush at all. Thanks!

bmcelvee commented on cli_reference/developer_cli_odo/understanding-odo.adoc:
```diff
@@ -121,3 +121,4 @@ wildfly     openshift     10.0,10.1,11.0,12.0,13.0,14.0,15.0,16.0,17.0,18.0,19.0
 ----
 +
 The `TAGS` column represents the available image versions, for example, `10` represents the `rhoar-nodejs/nodejs-10` container image.
+To learn more about the CLI commands, go to xref:../../cli_reference/openshift_developer_cli/odo-cli-reference.adoc#basic-odo-cli-commands_odo-cli-reference[odo CLI reference].
```
Apologies for the delay! I tinkered with the xref a bit and got it to work as follows: 

If you want to learn more about the CLI commands, see xref:../../cli_reference/developer_cli_odo/odo-cli-reference.adoc#basic-odo-cli-commands_odo-cli-reference[odo CLI reference].

jc-berger replied:
@bmcelvee, no worries about the "delay." Funnily enough, I was working on sending up a new commit while you were reviewing it. I noticed the same thing. 

I updated the XREF, and when I locally build it, it works for me. I also shortened the wording a little bit. 

Also, noted about the merge freeze. As long as it's merged next week, we should be fine.

jc-berger replied:
Comments addressed, so I've resolved the conversation.

bmcelvee commented:
Additional note: since we're currently in merge freeze, I removed the 4.6 label. I will re-label post-merge freeze.

bmcelvee commented:
Merging and cherry picking

bmcelvee commented:
/cherrypick enterprise-4.6

bmcelvee commented:
/cherrypick enterprise-4.7
